Simplify importing

As a user of this crate I just want to write `use hamcrest::*;` and import everything. However if I have a struct called `Matcher` that I want to test that won't work:

```
error[E0252]: a trait named `Matcher` has already been imported in this module
  --> src/patterns/small/test.rs:34:9
   |
24 | pub use hamcrest::*;
   |         ------------ previous import of `Matcher` here
...
34 | pub use super::Matcher;
   |         ^^^^^^^^^^^^^^ already imported

error: aborting due to previous error
```

I propose to add a submodule called `prelude` (like [rayon](https://github.com/nikomatsakis/rayon) does) so that we can use `use hamcrest::prelude::*;` instead. I'm proposing this as I think just _not_ exporting `Matcher` is not an option as this would mean that user's wouldn't be able to write their own custom matchers (is that the right term?).

--- a/README.md
+++ b/README.md
@@ -24,7 +24,11 @@ After a quick `cargo build`, you should be good to go!
 ## Usage
 
 Hamcrest supports a number of matchers. You'll have to `use`
-them just like any other Rust library.
+them just like any other Rust library. Alternatively you can also `use` all of them at once like this:
+
+``` rust
+use hamcrest::prelude::*;
+```
 
 ### equal\_to
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,6 +1,7 @@
 // Copyright 2014 Carl Lerche, Oliver Mader, Alex Crichton, Thiago Pontes,
 //                Yehuda Katz
 // Copyright 2015 Carl Lerche, Oliver Mader
+// Copyright 2016 Urban Hafner
 //
 // Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
 // http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
@@ -30,4 +31,15 @@ pub mod matchers {
     pub mod is;
     pub mod none;
     pub mod vecs;
+}
+
+pub mod prelude {
+    pub use core::assert_that;
+    pub use matchers::is::{is, is_not};
+    pub use matchers::is::is_not as not;
+    pub use matchers::none::{none};
+    pub use matchers::equal_to::equal_to;
+    pub use matchers::close_to::close_to;
+    pub use matchers::existing_path::{existing_path, existing_file, existing_dir};
+    pub use matchers::vecs::{of_len, contains};
 }

--- a/tests/prelude.rs
+++ b/tests/prelude.rs
@@ -1,0 +1,16 @@
+// Copyright 2016 Urban Hafner
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+extern crate hamcrest;
+
+use hamcrest::prelude::*;
+
+#[test]
+fn it_successfully_uses_imported_functions() {
+    assert_that(1, is(equal_to(1)));
+}

